torch/ao/sparsity/experimental/data_sparsifier/lightning/callbacks/data_sparsity.py

```diff
@@ -1,18 +1,27 @@
+from collections import defaultdict
 from copy import deepcopy
 import torch
 from dataclasses import dataclass, field
 from typing import Any, Optional, Dict
 import pytorch_lightning as pl
 
-from ._data_sparstity_utils import _attach_model_to_data_sparsifier, _log_sparsified_level
+from ._data_sparstity_utils import (
+    _attach_model_to_data_sparsifier,
+    _log_sparsified_level,
+    _create_data_scheduler,
+    _get_valid_name
+)
 
 
 @dataclass
```

Comment:
This is a little tricky: You are making it a dataclass, and you are inheriting from a callback -- this makes me concerned about the MRO. Also, AFAIK, the dataclass overrides the `__annotations__`, `__slots__`, and `__hash__`. imho, this might cause some problems if the Callbacks decide to use those.

Reply:
Yeah, this might be concerning. I've removed the data class and used a simple class with __init__() for now.